main/objpool.h

```diff
@@ -0,0 +1,44 @@
+/*
+*   Copyright (c) 2016, Jiri Techet
+*
+*   This source code is released for free distribution under the terms of the
+*   GNU General Public License version 2 or (at your option) any later version.
+*
+*   Defines generic pool for object reuse reducing the amount of allocations
+*   and deallocations.
+*/
+#ifndef CTAGS_MAIN_OBJPOOL_H
+#define CTAGS_MAIN_OBJPOOL_H
+
+/*
+*   INCLUDE FILES
+*/
+#include "general.h"  /* must always come first */
+
+#include "ptrarray.h"
+
+/*
+*   DATA DECLARATIONS
+*/
+typedef void * (*objPoolCreateFunc) (void);
+typedef void (*objPoolDeleteFunc) (void *data);
+typedef void (*objPoolClearFunc) (void *data);
+
+typedef struct sObjPool {
+	ptrArray *array;
+	unsigned int size;
+	objPoolCreateFunc createFunc;
+	objPoolDeleteFunc deleteFunc;
+	objPoolClearFunc clearFunc;
+} objPool;
+
```

Comment:
Could you move this to objpool.c?
If you could do it, you can remove ptrarray.h includsion, too.

(If circular reference of type definition is an issue in your work, main/types.h will help you.)